File: size_regression.py
import pyRserve
conn = pyRserve.connect()
conn.r.ls()

import pandas as pd
mydata = pd.read_csv("data/OutputData.csv")

coords= mydata.copy()
coords.drop(coords.columns[0:3],axis=1,inplace=True)
conn.r.coords=coords.to_numpy()

size=mydata.centeroid.to_numpy()
conn.r.size=size

conn.eval('require(geomorph)')
#conn.eval('require(Morpho)')
conn.eval('arr=arrayspecs(coords,p=41,k=3)')
conn.voidEval('gdf=geomorph.data.frame(Size=size,Coords=arr)')
conn.voidEval('outlm=procD.lm(Coords ~ Size,data=gdf)')
model=conn.eval('outlm')
#conn.eval('remove(list=ls())')
conn.shutdown 
#to shutdown Rserve. Otherwise it will stay open and contain all variable from the current session).

File: size_regression.R
library(SlicerMorphR)
library(geomorph)
#log=file.choose()

log="~/Desktop/2025-02-25_13_40_06/analysis.json"
parsed=parser2(log)

dat=read.csv(paste(parsed$output.path, parsed$OutputData, sep="/"))
size=dat$centroid
coords=arrayspecs(dat[,4:ncol(dat)], p=parsed$no.LM, k=3)
gdf=geomorph.data.frame(Size=size, Coords=coords)


size.model = as.formula(Coords~Size)
outlm=procD.lm(size.model, data=gdf)
